robotology/robots-configuration
# iCubGenova09 - Added xml files for feet F/T sensors

# DESCRIPTION

This PR adds the `xml` configuration files for dumping data from the `F/T` sensor assembled on the feet.

:information_source:  The other configuration files are the same of your local branch `iCubGenova09-devel`.

## CONNECTIONS

![soles](https://user-images.githubusercontent.com/6638215/99671065-ccc3ff00-2a71-11eb-9b72-1f2c050215b7.png)

| **Sensro placement**  | **CAN ID** |  **BOARD.CONNECTOR** |
|:--|:--|:-|
| Left Tiptoe | 13 | EB9.P4 |
| Left Heel | 13 | EB8.P5 |
| Right Tiptoe | 13 | EB13.P4 |
| Right Heel | 13 | EB12.P5 |

## YARP PORTS NAMING
I named the ports as follow : 

| **Sensro placement**  |  **Yarp Port Name** |
|:--|:--|
| Left Tiptoe | `/icub/left_foot_tiptoe/analog:o` `/icub/left_foot_tiptoe/rpc:o` |
| Left Heel | `/icub/left_foot_heel/analog:o` `/icub/left_foot_heel/rpc:o` |
| Right Tiptoe | `/icub/right_foot_tiptoe/analog:o` `/icub/right_foot_tiptoe/rpc:o` |
| Right Heel | `/icub/right_foot_heel/analog:o` `/icub/right_foot_heel/rpc:o` |


## TESTING
1. Run with dedicated `xml` file : 

` 
yarprobotinterface --config icub_FT_feet.xml
`

2. Verify that the ports have been created : 

![Screenshot from 2020-11-19 13-10-17](https://user-images.githubusercontent.com/6638215/99672817-3c3aee00-2a74-11eb-9834-bcf46fe38957.png)

3. Dump data from a port (all 4 ports have been tested) : 

![Screenshot from 2020-11-19 13-10-41](https://user-images.githubusercontent.com/6638215/99672969-71dfd700-2a74-11eb-8d90-f78ffa1ff80f.png)
